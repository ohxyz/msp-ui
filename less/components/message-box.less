.message-box {

    padding: 16px 15px 16px 50px;
    border: 1px solid #e3e3e3;
    position: relative;
    margin-bottom: 48px;
    font-size: 14px;

    .material-icons {

        position: absolute;
        top: 16px;
        left: 16px;
    }

    .message-box__title {

        margin-bottom: 16px;
        font-weight: bold;
        line-height: 24px;
    }

    .message-box__content {

        line-height: 24px;

        em {

            font-weight: bold;
            font-style: normal;
        }
    }

    &--success {

        background-color: #ebf6ec;
        border-color: #d6edd7;
        
        .material-icons {

            color: #007632;
        }

        .message-box__title {

            color: #007632;
        }

        .message-box__content {

            color: #007632;
        }
    }

    &--error {

        background-color: #feebeb;
        border-color: #fcd6d6;
        
        .material-icons {

            color: #ec0000;
        }

        .message-box__title {

            color: #7b0000;
        }

        .message-box__content {

            color: #7b0000;
        }
    }

    &--info {

        background-color: #f1f1f1;
        border-color: #e3e3e3;
        
        .material-icons {

            color: #505050;
        }

        .message-box__title {

            color: #505050;
        }

        .message-box__content {

            color: #505050;
        }  
    }
}