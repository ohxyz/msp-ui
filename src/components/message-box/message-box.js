const util = require( '../../helpers/util.js' );
const React = require( 'react' );

const DEFAULT_ICON_STYLE = 'info';
const DEFAULT_SECONDS_TO_DISMISS = -1;

class MessageBox extends React.Component {

    constructor( props ) {

        super( props );

        this.state = {

            type: '',
            iconStyle: '',
            title: '',
            content: '',
            shouldDisplay: true,
            secondsToDismiss: 0,
            onPropsDismiss: new Function()
        };

        this.mapOfTypeAndIcon = {

            'warning': 'warning',
            'error': 'cancel',
            'success': 'check_circle',
            'info': 'info',
        };
    }

    static getDerivedStateFromProps( nextProps, prevState ) {
        
        return {

            type: util.setDefault( nextProps.type, '' ),
            iconStyle: util.setDefault( nextProps.iconStyle, '' ),
            title: util.setDefault( nextProps.title, '' ),
            content: util.setDefault( nextProps.content, '' ),
            shouldDisplay: util.setDefault( nextProps.shouldDisplay, true ),
            secondsToDismiss: util.setDefault( nextProps.seconds, DEFAULT_SECONDS_TO_DISMISS ),
            onPropsDismiss: util.setDefault( nextProps.onDismiss, new Function() )
        };
    }

    dismissInSeconds( seconds ) {

        let milliseconds = parseInt( seconds, 10 ) * 1000;

        setTimeout( () => { 

            this.setState( {

                shouldDisplay: false

            } );

            this.state.onPropsDismiss();
        
        }, milliseconds );
    }

    componentDidMount() {

        let seconds = this.state.secondsToDismiss;

        if ( seconds > 0 ) {

            this.dismissInSeconds( seconds );
        }
    }

    render() {

        if ( this.state.shouldDisplay === false ) {

            return null;
        }

        let className = 'message-box';
        let iconLiteral = this.state.iconStyle;
        let messageType = this.state.type;
        let defaultIconStyle = 'info';

        if ( iconLiteral === '' ) {

            iconLiteral = this.mapOfTypeAndIcon[ messageType ];

            if ( iconLiteral === undefined ) {

                iconLiteral = DEFAULT_ICON_STYLE;
            }
        }

        if ( messageType !== '' ) {

            className += ' message-box--' + messageType;
        }
        else {

            className += ' message-box--info';
        }
        
        return (

            <div className={ className }>
                <i className="material-icons">{ iconLiteral }</i>
                <div className="message-box__title">{ this.state.title }</div>
                <div className="message-box__content">{ this.state.content }</div>
            </div>
        );
    }
}

export {

    MessageBox,
};