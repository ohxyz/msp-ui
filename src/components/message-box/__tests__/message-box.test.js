import React from 'react';
import { MessageBox } from '../message-box.js';
import renderer from 'react-test-renderer';

it( 'should render with defaults', () => { 

    const tree = renderer
        .create( <MessageBox title="Hi" content="Hello there!"/> )
        .toJSON();

    expect( tree ).toMatchSnapshot();

} );

it( 'should render by iconStyle when type is also there', () => { 

    const tree = renderer
        .create( <MessageBox iconStyle="info" type="warning" title="Hi" content="Hello there!"/> )
        .toJSON();

    expect( tree ).toMatchSnapshot();

} );