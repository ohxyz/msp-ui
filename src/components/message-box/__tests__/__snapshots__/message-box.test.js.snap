// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render by iconStyle when type is also there 1`] = `
<div
  className="message-box message-box--warning"
>
  <i
    className="material-icons"
  >
    info
  </i>
  <div
    className="message-box__title"
  >
    Hi
  </div>
  <div
    className="message-box__content"
  >
    Hello there!
  </div>
</div>
`;

exports[`should render with defaults 1`] = `
<div
  className="message-box message-box--info"
>
  <i
    className="material-icons"
  >
    info
  </i>
  <div
    className="message-box__title"
  >
    Hi
  </div>
  <div
    className="message-box__content"
  >
    Hello there!
  </div>
</div>
`;
